std/meta.zig

```diff
@@ -117,6 +117,44 @@ test "std.meta.Child" {
     testing.expect(Child(promise->u8) == u8);
 }
 
+/// Given an array/pointer type, return the slice type `[]Child`. Preserves `const`.
```

Comment:
what about the rest of the pointer properties?
        is_volatile: bool,
        alignment: comptime_int,
        child: type,
        is_allowzero: bool,

Reply:
I started implementing this and realized that supporting all those properties with the way I've done this is going to result in a binary exponential explosion of code.  Since every combination of properties needs to drill down to an explicit type.  For `N` properties we have `2^N` cases to handle.  So to support 3 properties, we have `8` cases to handle, i.e.
```zig
    if (info.is_const) {
        if (info.is_volatile) {
            if (info.is_allowzero)
                return []const volatile allowzero T;
            else
                return []const volatile T;
        } else {
            if (info.is_allowzero)
                return []const allowzero T;
            else
                return []const T;
        }
    } else {
        if (info.is_volatile) {
            if (info.is_allowzero)
                return []volatile allowzero T;
            else
                return []volatile T;
        } else {
            if (info.is_allowzero)
                return []allowzero T;
            else
                return [] T;
        }
    }
```

And we would have to duplicate this code for every type modifier function such as `SliceType`, `ArrayPointerType` and any functions we deem suitable for the standard library.

This doesn't seem to scale. One scalable solution would be to construct the types programatically.  The above example could be done with something like this:
```zig
    return @typeFromInfo(@import("builtin").TypeInfo.Pointer {
        .is_const = info.is_const,
        .is_volatile = info.is_volatile,
        .is_allowzero = info.is_allowzero,
    });
```

However I don't think the `@typeFromInfo` function exists.  Is this something that could be added?
```zig
@typeInfo(comptime info: @import("builtin").TypeInfo) type
```